[Kakamotobi] Week 3

Kakamotobi opened the pull request.

Kakamotobi committed:
Create MainNews component

- The `MainNews` component observes the `MainContentView` state ("list-view" or "grid-view") and is responsible for rendering the `GridView` or `ListView` component based on the state.
- `GridView` and `ListView` will observe `gridViewData` and `listViewData`, respectively.

Kakamotobi committed:
Add GridView component

- Connect list/grid view buttons to MainContentView state in MainNews.
- Render GridView in MainNews when the MainContentView state is set to "main-view".
- Add scss variables for svg filters.

Kakamotobi committed:
Change method name from update to setProps

Kakamotobi committed:
Update dev log

Kakamotobi committed:
Fix grid view crawler and update crawled data

- Ensure that the page is on the entire press view of the newsstand tab.

saejinpark commented on src/store/index.ts:
```diff
@@ -2,44 +2,66 @@ import Component from "@components/common/Component.ts";
-  gridView: [];
-  listView: [];
+  recentHeadlinesData: { pressName: string; headlineTitle: string }[];
+  gridViewData: { src: string; alt: string }[];
```
{ pressName: string; headlineTitle: string } 이나 { src: string; alt: string } 이 부분도 interface로 만들어주면 어떨까요?

Kakamotobi replied:
다른 곳에도 사용하고 있던걸로 기억하는데 따로 선언해주는게 좋을것 같습니다.👍

saejinpark commented on src/components/MainNews/GridView.scss:
```diff
@@ -0,0 +1,75 @@
+    border: none;
+
+    &.left-btn {
+      left: -74px;
```
rem을 사용하면 텍스트에 상대적인 크기를 사용하여 많은 이점이 있습니다. 혹시 시간이 되신다면 한번 전환하여 사용해보는 것도 좋을것 같습니다.

saejinpark commented on src/components/MainNews/MainNews.ts:
```diff
@@ -0,0 +1,83 @@
+    super(topElement);
+    this.contentViewBtnsWrapper = contentViewBtnsWrapper;
+    this.main = main;
+
```
이런 부분이 너무 긴것 같은 생각이 들어서 컴포넌트를 좀 더 생성하여 활용해보시면 어떨까요

saejinpark reviewed (commented):
코드잘봤습니다. 제가 타입스크립트가 미숙해서 리뷰가 많이 부족합니다.
주말에 크롱이 피드백을 주었던 클래스를 줄이는 방법을 고민하면서 봤던 영상인데 좋은 것 같아 공유드립니다. 
https://www.youtube.com/watch?v=SuTITWxw-Gw&t=208s

Kakamotobi committed:
Change property name from title to categoryName

Kakamotobi committed:
Add list view store

Kakamotobi committed:
Add events directly to content view buttons rather than event delegation

- Seems unnecessary to delegate the event when there are only two possible targets.

Kakamotobi committed:
Add primary faded color

Kakamotobi committed:
Restructure store

- Combine stores and their corresponding collection of observers together.

Kakamotobi committed:
Add path alias for types folder

Kakamotobi committed:
Seperate shared types into separate folder

- If used in two or more places, place in types folder.
- If used in only one place, place at the site.

Kakamotobi committed:
Refactor reducer and create type for list view props

- Separate reducer logic into handlers.
